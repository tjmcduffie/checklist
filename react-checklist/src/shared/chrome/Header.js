import classnames from 'classnames';
import React from 'react';

const Header = ({onToggleShowCompleted, shouldShowCompleted}) => {
  return (
    <header className={classnames('flex-row')}>
      <h1 className={classnames('flex-grow')}>Checklist</h1>
      <button
        className={classnames('clear-button', 'flex-shrink', 'flex-align-center')}
        onClick={onToggleShowCompleted}
        onKeyPress={
          e => ((e.key === " " || e.key === "Enter") && onToggleShowCompleted())
        }>
        {shouldShowCompleted ? 'Hide completed' : 'Show completed'}
      </button>
    </header>
  );
};

export default Header;
