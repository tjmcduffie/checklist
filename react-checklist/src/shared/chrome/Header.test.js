import {shallow} from 'enzyme';
import Header from './Header';
import React from 'react';
import {create} from 'react-test-renderer';


import {Link} from 'react-router-dom';

jest.mock('react-router-dom', () => ({
  Link: 'Link'
}));

describe('Header', () => {
  const noop = () => {};

  it('renders without crashing', () => {
    const elem = create(<Header />);
    elem.unmount();
  });

  describe('renders predictably', () => {
    it('when showing completed items', () => {
      const elem = create(<Header />);
      expect(elem.toJSON()).toMatchSnapshot();
    });
    it('when hiding completed items', () => {
      const elem = create(<Header shouldShowCompleted={false} />);
      expect(elem.toJSON()).toMatchSnapshot();
    });
  });

  describe('renders predictably', () => {
    it('triggers the toggle show completed callback', () => {
      const mockOnToggle = jest.fn();
      const elem = shallow(<Header onToggleShowCompleted={mockOnToggle} />);
      const button = elem.find('button');

      button.simulate('click');
      expect(mockOnToggle.mock.calls.length).toBe(1);

      button.simulate('keypress', {key: ' '});
      expect(mockOnToggle.mock.calls.length).toBe(2);

      button.simulate('keypress', {key: 'Enter'});
      expect(mockOnToggle.mock.calls.length).toBe(3);

      button.simulate('keypress', {key: 'TAB'});
      expect(mockOnToggle.mock.calls.length).toBe(3);
    });
  });
});
