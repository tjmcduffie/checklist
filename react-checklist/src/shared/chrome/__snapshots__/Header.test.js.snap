// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders predictably when hiding completed items 1`] = `
<header
  className="flex-row"
>
  <h1
    className="flex-grow"
  >
    Checklist
  </h1>
  <button
    className="clear-button flex-shrink flex-align-center"
    onKeyPress={[Function]}
  >
    Show completed
  </button>
</header>
`;

exports[`Header renders predictably when showing completed items 1`] = `
<header
  className="flex-row"
>
  <h1
    className="flex-grow"
  >
    Checklist
  </h1>
  <button
    className="clear-button flex-shrink flex-align-center"
    onKeyPress={[Function]}
  >
    Show completed
  </button>
</header>
`;
